src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,259 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String createCarRequest = "INSERT into cars (manufacturer_id, model) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(createCarRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, car.getManufacturer().getId());
+            statement.setString(2, car.getModel());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+            addDriversToCarsDriversTable(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create Car: " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getCarRequest =
+                "SELECT c.id AS car_id, c.model, c.manufacturer_id "
+                + "FROM cars c "
+                + "WHERE TRUE "
+                + "AND c.id = ? "
+                + "AND c.is_deleted = 0 ";
+        Car car = new Car();
+        ResultSet resultSet;
+        Long driverId = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(getCarRequest,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, id);
+            resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car.setId(id);
+                car.setModel(resultSet.getString("model"));
```

Comment:
could we move this logic to separate method?

Reply:
1. there are only two duplicate lines of code, I think it would be more confusing to create a separate method to instantiate a car and set it's model, then to leave it as it is.